hibernate-core/src/main/java/org/hibernate/cache/internal/NoCachingTransactionSynchronizationImpl.java

```diff
@@ -6,14 +6,35 @@
  */
 package org.hibernate.cache.internal;
 
-import org.hibernate.cache.spi.AbstractCacheTransactionSynchronization;
-import org.hibernate.cache.spi.RegionFactory;
+import org.hibernate.cache.spi.CacheTransactionSynchronization;
 
 /**
  * @author Steve Ebersole
  */
-public class NoCachingTransactionSynchronizationImpl extends AbstractCacheTransactionSynchronization {
-	public NoCachingTransactionSynchronizationImpl(RegionFactory regionFactory) {
-		super( regionFactory );
+public class NoCachingTransactionSynchronizationImpl implements CacheTransactionSynchronization {
+	public static final NoCachingTransactionSynchronizationImpl INSTANCE = new NoCachingTransactionSynchronizationImpl();
+
+	private NoCachingTransactionSynchronizationImpl() {
+
+	}
+
+	@Override
+	public long getCachingTimestamp() {
+		throw new UnsupportedOperationException("No Caching does not support Timestamp generation");
```

Comment:
I'm not sure if it's clear what "No Caching" means. Could you clarify that?

Reply:
changed, what do you think? any better?